book/src/todo_3.md

```diff
@@ -0,0 +1,205 @@
+
+Now, if we go to the settings and request a different style, our app will oblige.
+
+<div style="text-align:center">
+ <video autoplay muted loop>
+  <source src="vid/todo_dark.webm" type="video/webm">
+Your browser does not support the video tag.
+ </video>
+</div>
+
+
+## Start using Libadwaita widgets
+
+Of course Libadwaita is more than just a couple of stylesheets and a [`StyleManager`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.StyleManager.html).
+But before we get to the interesting stuff, we make our lives easier in the future by replacing all occurrences of `gtk::prelude` and `gtk::subclass::prelude` with [`adw::prelude`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/prelude/index.html) and [`adw::subclass::prelude`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/subclass/prelude/index.html).
+This works because the `adw` preludes re-export the corresponding `gtk` preludes plus a couple of Libadwaita specific traits.
+
+In the remainder of this chapter we are going to follow a couple of patterns of GNOME's HIG.
+Let's start by adapting the header bar to be [HIG compliant](https://developer.gnome.org/hig/patterns/containers/header-bars.html).
+We do that by replacing [`gtk::ApplicationWindow`](../docs/gtk4/struct.ApplicationWindow.html) with [`adw::ApplicationWindow`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.ApplicationWindow.html).
+We also use [`adw::HeaderBar`](https://world.pages.gitlab.gnome.org/Rust/libadwaita-rs/stable/latest/docs/libadwaita/struct.HeaderBar.html) as title bar.
+Finally, we add [tooltips](https://developer.gnome.org/hig/patterns/feedback/tooltips.html).
+
+The second pattern we are going to follow are [boxed lists](https://developer.gnome.org/hig/patterns/containers/boxed-lists.html).
+The HIG does not require us to use them and there's a good reason for that: they are incompatible with recycling lists.
+This means they cannot be used with [list views](https://developer.gnome.org/hig/patterns/containers/list-column-views.html) and are therefore only appropriate for relatively small lists.
+
+> Try to add tasks programmatically and see how many of them you have to add until the UI noticeably slows down.
+> Determine for yourself if you think that is a reasonable number or if we should have rather stuck with list views.
+
+We can use boxed lists by using [`gtk::ListBox`](../docs/gtk4/struct.ListBox.html) instead of [`gtk::ListView`](../docs/gtk4/struct.ListView.html).
+Afterwards, we match with the [`boxed-list`](https://gnome.pages.gitlab.gnome.org/libadwaita/doc/main/boxed-lists.html) style class that is provided by Libadwaita.
+
+Let's implement the discussed changes in the `window.ui` file.
+You can find the relevant subset of the diff below.
+To see the complete file, just click on the link after "Filename:".
```

Comment:
Is this needed? Like, it's chapter 15 already :)

Reply:
Fair, but I only mention it once in the beginning of the book and this is the first time that I show an incomplete diff